libs/api_common/device_helper.cpp

```diff
@@ -26,22 +26,40 @@ typedef struct _async_ioctl_completion_context
     async_ioctl_completion_callback_t callback;
 } async_ioctl_completion_context_t;
 
-static ebpf_handle_t _device_handle = ebpf_handle_invalid;
+// Handle used for synchronous calls to the driver.
+static ebpf_handle_t _sync_device_handle = ebpf_handle_invalid;
```

Comment:
do we need a handle per thread (TLS)? else all the sync IOCTL calls will be serialized.

Reply:
I tried that approach, but it runs into issues with our existing unit testing framework. Specifically, the TLS destroy occurs after the test completes and the closehandle mock has been remove. 

My proposed solution is to fix this the simple way for now, and then take the time to build a completer and more performant fix when we are not time bound.

Reply:
I would recommend to modify `ebpf_clear_thread_local_storage` to keep the handle thing alive?

Reply:
Lets commit this as v1. We can then decide on a more performant v2.